### DISCUSSION TITLE
How to utilize conversational context in Dynamic Routes?

### DISCUSSION THREAD

**Discussion by agyaatcoder** [aurelio-labs/semantic-router#245] [Q&A]
How to utilize conversational context and knowledge, especially in multi-turn or chat contexts, for  dynamic routes? For example, if a user asks: “How's is the weather there?”, route layer is classified correctly, but the function called is get_weather("there"). As the routing is applied at the query level, how to utilize contextual info from chat history to format corresponding API calls?

**Comment by octocat20** [comment]
Dynamic routes only see the string you pass in. They do not pull chat history for you, so "How's the weather there?" becomes get_weather("there").

Ground the turn first, then route:

- rewrite with history ("there" -> "Berlin") and call the router on that
- or prepend a short context blurb to the query you send in
- or keep simple app state (last city, etc.) and fill missing args after you get the RouteChoice

Router picks the route and can extract args from one utterance. Multi-turn grounding is on you.
